Build Issue with the develop branch

I was trying to build the develop branch with a machine with CUDA 12 and GCC 11. I am getting the following errors during the compilation:

```
[  3%] Building CXX object src/CMakeFiles/hykkt_libs.dir/MMatrix.cpp.o
[  6%] Building CXX object src/CMakeFiles/hykkt_libs.dir/input_functions.cpp.o
[ 10%] Building CXX object src/CMakeFiles/hykkt_libs.dir/vector_vector_ops.cpp.o
[ 13%] Building CXX object src/CMakeFiles/hykkt_libs.dir/cusparse_utils.cpp.o
In file included from /localwork/fda/ExaSGD/hykkt/src/matrix_vector_ops.hpp:5,
                 from /localwork/fda/ExaSGD/hykkt/src/cusparse_utils.cpp:7:
/usr/local/cuda/targets/x86_64-linux/include/cublas_v2.h:59:2: error: #error "It is an error to include both cublas.h and cublas_v2.h"
   59 | #error "It is an error to include both cublas.h and cublas_v2.h"
      |  ^~~~~
make[2]: *** [src/CMakeFiles/hykkt_libs.dir/build.make:118: src/CMakeFiles/hykkt_libs.dir/cusparse_utils.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:128: src/CMakeFiles/hykkt_libs.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

--- a/.github/workflows/push_mirror.yaml
+++ b/.github/workflows/push_mirror.yaml
@@ -1,7 +1,7 @@
 name: Mirroring
 
 # triggers a github action everytime there is a push or mr
-on: [push]
+# on: [push]
 
 jobs:
   # To test on HPC resources we must first mirror the repo and then trigger a pipeline

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,9 +8,9 @@ set(CMAKE_CXX_STANDARD 11)
 set(PACKAGE_NAME  "HyKKT")
 set(PACKAGE_TARNAME "hykkt")
 
-option(HYKKT_TEST_WITH_BSUB "Use `jsrun` instead of `mpirun` commands when running tests" OFF)
 option(HYKKT_USE_AMD "Use AMD from SuiteSparse" OFF)
 set(HYKKT_CTEST_OUTPUT_DIR ${PROJECT_BINARY_DIR} CACHE PATH "Directory where CTest outputs are saved")
+set(HYKKT_RUN_CMD "" CACHE STRING "Separate strings by semicolons, e.g. compute-sanitizer;--tool;memcheck")
 
 set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}")
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)

--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -87,8 +87,8 @@ set(test_solver2_args
 
 if(HYKKT_TEST_WITH_BSUB)
   set(RUNCMD "jsrun" "-n" "1" "-g" "1")
-else()
-  set(RUNCMD "cuda-memcheck") # No special command is needed to run this program  
+# else()
+#   set(RUNCMD "cuda-memcheck") # No special command is needed to run this program  
 endif()
 
 add_executable(perm_driver perm_driver.cpp)

--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -85,45 +85,40 @@ set(test_solver2_args
   3
   10000.0)
 
-if(HYKKT_TEST_WITH_BSUB)
-  set(RUNCMD "jsrun" "-n" "1" "-g" "1")
-# else()
-#   set(RUNCMD "cuda-memcheck") # No special command is needed to run this program  
-endif()
 
 add_executable(perm_driver perm_driver.cpp)
 target_link_libraries(perm_driver PUBLIC HyKKT::HyKKT)
 add_test(NAME permutation
-  COMMAND ${RUNCMD} $<TARGET_FILE:perm_driver>)
+  COMMAND ${HYKKT_RUN_CMD} $<TARGET_FILE:perm_driver>)
 
 add_executable(ruiz_driver Ruiz_driver.cpp)
 target_link_libraries(ruiz_driver PUBLIC HyKKT::HyKKT)
 add_test(NAME ruiz_scaling_test
-  COMMAND ${RUNCMD} $<TARGET_FILE:ruiz_driver>)
+  COMMAND ${HYKKT_RUN_CMD} $<TARGET_FILE:ruiz_driver>)
 
 add_executable(chol_driver cuSolver_driver_cholesky.cpp)
 target_link_libraries(chol_driver PUBLIC HyKKT::HyKKT)
 add_test(NAME cholesky 
- COMMAND ${RUNCMD} $<TARGET_FILE:chol_driver> ${testchol_args})
+ COMMAND ${HYKKT_RUN_CMD} $<TARGET_FILE:chol_driver> ${testchol_args})
 
 add_executable(schur_cg_driver cuSolver_driver_schur_cg.cpp)
 target_link_libraries(schur_cg_driver PUBLIC HyKKT::HyKKT)
 add_test(NAME schur_cg 
-  COMMAND ${RUNCMD} $<TARGET_FILE:schur_cg_driver> ${testcg_args})
+  COMMAND ${HYKKT_RUN_CMD} $<TARGET_FILE:schur_cg_driver> ${testcg_args})
 
 add_executable(hybrid_driver cuSolver_driver_hybrid.cpp)
 target_link_libraries(hybrid_driver PUBLIC HyKKT::HyKKT)
 add_test(NAME hybrid1 
-  COMMAND ${RUNCMD} $<TARGET_FILE:hybrid_driver> ${test1_args})
+  COMMAND ${HYKKT_RUN_CMD} $<TARGET_FILE:hybrid_driver> ${test1_args})
 add_test(NAME hybrid2
-  COMMAND ${RUNCMD} $<TARGET_FILE:hybrid_driver> ${test2_args})
+  COMMAND ${HYKKT_RUN_CMD} $<TARGET_FILE:hybrid_driver> ${test2_args})
 
 add_executable(solver_driver cuSolver_driver_solver.cpp)
 target_link_libraries(solver_driver PUBLIC HyKKT::HyKKT)
 add_test(NAME solver1
-  COMMAND ${RUNCMD} $<TARGET_FILE:solver_driver> ${test_solver1_args})
+  COMMAND ${HYKKT_RUN_CMD} $<TARGET_FILE:solver_driver> ${test_solver1_args})
 add_test(NAME solver2
-  COMMAND ${RUNCMD} $<TARGET_FILE:solver_driver> ${test_solver2_args})
+  COMMAND ${HYKKT_RUN_CMD} $<TARGET_FILE:solver_driver> ${test_solver2_args})
   
 add_executable(solver_driver_emulator cuSolver_driver_solver_emulator.cpp)
 target_link_libraries(solver_driver_emulator PUBLIC HyKKT::HyKKT)

--- a/src/HykktSolver.cpp
+++ b/src/HykktSolver.cpp
@@ -1,7 +1,7 @@
 #include <cassert>
 
 #include <cusparse.h>
-#include <cublas.h>
+#include <cublas_v2.h>
 
 #include "HykktSolver.hpp"
 

--- a/src/HykktSolver.hpp
+++ b/src/HykktSolver.hpp
@@ -1,7 +1,7 @@
 #pragma once
 
 #include <cusparse.h>
-#include <cublas.h>
+#include <cublas_v2.h>
 
 #include <cuda_memory_utils.hpp>
 #include "MMatrix.hpp"

--- a/src/SchurComplementConjugateGradient.hpp
+++ b/src/SchurComplementConjugateGradient.hpp
@@ -1,7 +1,7 @@
 #pragma once
 
 #include <cusparse.h>
-#include <cublas.h>
+#include <cublas_v2.h>
 #include <cusparse_utils.hpp>
 
 // Froward declaration of CholeskyClass

--- a/src/cusparse_utils.cpp
+++ b/src/cusparse_utils.cpp
@@ -1,5 +1,5 @@
 #include <cusparse.h>
-#include <cublas.h>
+#include <cublas_v2.h>
 
 #include <cusparse_utils.hpp>
 #include "cuda_memory_utils.hpp"

--- a/src/matrix_vector_ops.cu
+++ b/src/matrix_vector_ops.cu
@@ -83,7 +83,7 @@ void fun_SpMV_buffer(cusparseHandle_t handle,
                                           &beta, 
                                           c_desc_dn, 
                                           COMPUTE_TYPE, 
-                                          CUSPARSE_MV_ALG_DEFAULT, 
+                                          CUSPARSE_SPMV_ALG_DEFAULT, 
                                           buffer_size));
 }
 
@@ -103,7 +103,7 @@ void fun_SpMV_product(cusparseHandle_t handle,
                                &beta, 
                                c_desc_dn, 
                                COMPUTE_TYPE, 
-                               CUSPARSE_MV_ALG_DEFAULT, 
+                               CUSPARSE_SPMV_ALG_DEFAULT, 
                                buffer));
 }
 

